src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -354,6 +354,54 @@ begin
     exact ⟨_, _, h.2 ha, rfl, rfl⟩ }
 end
 
+/-- Graph homomorphisms induce a contravariant function on subgraphs. -/
+@[simps]
+protected def comap {G' : simple_graph W} (f : G →g G') (H : G'.subgraph) : G.subgraph :=
+{ verts := f ⁻¹' H.verts,
+  adj := λ u v, G.adj u v ∧ H.adj (f u) (f v),
+  adj_sub := by { rintros v w ⟨ga, ha⟩, exact ga },
+  edge_vert := by { rintros v w ⟨ga, ha⟩, simp [H.edge_vert ha] } }
+
+lemma comap_monotone {G' : simple_graph W} (f : G →g G') : monotone (subgraph.comap f) :=
+begin
+  intros H H' h,
+  split,
+  { intro,
+    simp only [comap_verts, set.mem_preimage],
+    apply h.1, },
+  { intros v w,
+    simp only [comap_adj, and_imp, true_and] { contextual := tt },
+    intro,
+    apply h.2, }
+end
+
+lemma map_le_iff_le_comap {G' : simple_graph W} (f : G →g G') (H : G.subgraph) (H' : G'.subgraph) :
+  H.map f ≤ H' ↔ H ≤ H'.comap f :=
+begin
+  split,
```

Comment:
This proof is long enough that I think it's acceptable to compress it as such:
```suggestion
  refine ⟨λ h, ⟨λ v, _, λ v w hvw, _⟩, λ h, ⟨λ v, _, λ v w, _⟩⟩,
```
as that's already 8 lines less.

Reply:
Thanks